packages/toolkit/src/useEnvironment.ts

```diff
@@ -90,74 +86,39 @@ function configureApm(
   }
 }
 
-function configureProvider(
-  network: string,
-  truffleNetwork: {
-    provider: any
-    host: string
-    port: number
-  },
-  useFrame: boolean
-): any | WebsocketProvider | HttpProvider {
-  if (useFrame) {
-    return new Web3.providers.WebsocketProvider(
-      FRAME_ENDPOINT,
-      { headers: { origin: FRAME_ORIGIN } } // Provider options
-    )
-  }
-
-  if (!truffleNetwork) {
-    throw new NoNetworkInTruffleConfig(network)
-  }
-
-  const { provider, host, port } = truffleNetwork
-
-  if (provider) {
-    if (typeof provider === 'function') return provider()
-    else return provider
-  } else if (host && port) {
-    return new Web3.providers.WebsocketProvider(`ws://${host}:${port}`)
-  } else {
-    return new Web3.providers.HttpProvider('http://localhost:8545')
-  }
-}
-
 function configureEthersProvider({
```

Comment:
The Toolkit could be refactored to:
1. Ask for an initialized provider as Input
2. Return data or a transaction object to sign